File: Tests/Tests/STPPaymentMethodiDEALTest.m
//
//  STPPaymentMethodiDEALTest.m
//  StripeiOS Tests
//
//  Created by Yuki Tokuhiro on 3/9/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

#import <XCTest/XCTest.h>

#import "STPPaymentMethodIdeal.h"

@interface STPPaymentMethodiDEALTest : XCTestCase

@end

@implementation STPPaymentMethodiDEALTest

- (NSDictionary *)exampleJson {
    return @{
             @"bank": @"Rabobank",
             @"bic": @"RABONL2U",
             };
}

- (void)testDecodedObjectFromAPIResponseRequiredFields {
    NSArray<NSString *> *requiredFields = @[];
    
    for (NSString *field in requiredFields) {
        NSMutableDictionary *response = [[self exampleJson] mutableCopy];
        [response removeObjectForKey:field];
        
        XCTAssertNil([STPPaymentMethodiDEAL decodedObjectFromAPIResponse:response]);
    }
    
    XCTAssert([STPPaymentMethodiDEAL decodedObjectFromAPIResponse:[self exampleJson]]);
}

- (void)testDecodedObjectFromAPIResponseMapping {
    NSDictionary *response = [self exampleJson];
    STPPaymentMethodiDEAL *ideal = [STPPaymentMethodiDEAL decodedObjectFromAPIResponse:response];
    XCTAssertEqualObjects(ideal.bankName, @"Rabobank");
    XCTAssertEqualObjects(ideal.bankIdentifierCode, @"RABONL2U");
}

@end

File: Stripe/StripeiOSTests/STPPaymentMethodiDEALTest.swift
//  Converted to Swift 5.8.1 by Swiftify v5.8.28463 - https://swiftify.com/
//
//  STPPaymentMethodiDEALTest.m
//  StripeiOS Tests
//
//  Created by Yuki Tokuhiro on 3/9/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

class STPPaymentMethodiDEALTest: XCTestCase {
    func exampleJson() -> [AnyHashable: Any]? {
        return [
            "bank": "Rabobank",
            "bic": "RABONL2U",
        ]
    }

    func testDecodedObjectFromAPIResponseRequiredFields() {
        let requiredFields: [String]? = []

        for field in requiredFields ?? [] {
            var response = exampleJson()
            response?.removeValue(forKey: field)

            XCTAssertNil(STPPaymentMethodiDEAL.decodedObject(fromAPIResponse: response))
        }

        XCTAssertNotNil(STPPaymentMethodiDEAL.decodedObject(fromAPIResponse: exampleJson()))
    }

    func testDecodedObjectFromAPIResponseMapping() {
        let response = exampleJson()
        let ideal = STPPaymentMethodiDEAL.decodedObject(fromAPIResponse: response)
        XCTAssertEqual(ideal?.bankName, "Rabobank")
        XCTAssertEqual(ideal?.bankIdentifierCode, "RABONL2U")
    }
}
